golem-worker-service-base/src/evaluator/evaluator_context.rs

```diff
@@ -83,15 +127,47 @@ impl EvaluationContext {
     pub fn from_all(
         worker_detail: &WorkerDetail,
         request: &RequestDetails,
-        functions: Vec<AnalysedFunction>,
+        component_metadata: ComponentMetadata,
     ) -> Self {
         let mut request_data = internal::request_type_annotated_value(request);
         let worker_data = create_record("worker", worker_detail.clone().to_type_annotated_value());
         let merged = request_data.merge(&worker_data);
 
+        let top_level_functions = component_metadata.functions();
```

Comment:
This is all a tremendous amount of work to do on each invocation. In the future we need to cache this.